plugins/npa-v1alpha2/clusternetworkpolicy.go

```diff
@@ -0,0 +1,273 @@
+	logger.V(2).Info("Egress CNP evaluation", "tier", c.tier, "npolicies", len(policies), "action", action)
+
+	return actionToVerdict(action), nil
+}
+
+// getPoliciesForPod filters policies from the lister that match the pod and the evaluator's tier.
+func (c *ClusterNetworkPolicy) getPoliciesForPod(pod *api.PodInfo) ([]*npav1alpha2.ClusterNetworkPolicy, error) {
+	if pod == nil {
+		return nil, nil
+	}
+
+	allPolicies, err := c.cnpLister.List(labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+
+	var result []*npav1alpha2.ClusterNetworkPolicy
+	for _, policy := range allPolicies {
+		// Filter by the tier this evaluator instance is responsible for.
+		if policy.Spec.Tier != c.tier {
+			continue
+		}
+
+		subject := policy.Spec.Subject
+		matches := false
+		if subject.Namespaces != nil && networkpolicy.MatchesSelector(subject.Namespaces, pod.Namespace.Labels) {
+			matches = true
+		}
+		if !matches && subject.Pods != nil &&
+			networkpolicy.MatchesSelector(&subject.Pods.NamespaceSelector, pod.Namespace.Labels) &&
+			networkpolicy.MatchesSelector(&subject.Pods.PodSelector, pod.Labels) {
+			matches = true
+		}
+
+		if matches {
+			result = append(result, policy)
+		}
+	}
+
+	// Sort by priority and then by name for deterministic evaluation.
+	slices.SortFunc(result, func(a, b *npav1alpha2.ClusterNetworkPolicy) int {
+		if n := cmp.Compare(a.Spec.Priority, b.Spec.Priority); n != 0 {
+			return n
+		}
+		return cmp.Compare(a.Name, b.Name)
+	})
+	return result, nil
+}
+
+// evaluateClusterEgress evaluates a list of egress policies for a traffic flow.
+func (c *ClusterNetworkPolicy) evaluateClusterEgress(
+	policies []*npav1alpha2.ClusterNetworkPolicy,
+	dstPod *api.PodInfo,
+	dstIP net.IP,
+	dstPort int,
+	protocol v1.Protocol,
+) npav1alpha2.ClusterNetworkPolicyRuleAction {
+	for _, policy := range policies {
+		for _, rule := range policy.Spec.Egress {
+			if rule.Ports != nil {
+				if !evaluateClusterNetworkPolicyPort(*rule.Ports, dstPod, dstPort, protocol) {
+					continue
+				}
+			}
+			for _, to := range rule.To {
+				if to.Namespaces != nil && dstPod != nil && networkpolicy.MatchesSelector(to.Namespaces, dstPod.Namespace.Labels) {
+					return rule.Action
+				}
+
+				if to.Pods != nil && dstPod != nil &&
+					networkpolicy.MatchesSelector(&to.Pods.NamespaceSelector, dstPod.Namespace.Labels) &&
+					networkpolicy.MatchesSelector(&to.Pods.PodSelector, dstPod.Labels) {
+					return rule.Action
+				}
+
+				if to.Nodes != nil && dstPod != nil && networkpolicy.MatchesSelector(to.Nodes, dstPod.Node.Labels) {
+					return rule.Action
+				}
+
+				for _, network := range to.Networks {
+					_, cidr, err := net.ParseCIDR(string(network))
+					if err != nil {
+						continue
+					}
+					if cidr.Contains(dstIP) {
+						return rule.Action
+					}
+				}
+				for _, domain := range to.DomainNames {
+					if c.domainResolver != nil && c.domainResolver.ContainsIP(string(domain), dstIP) {
```

Comment:
I think domainResolver is only nil for unit tests, but should we log something anyway in this case?